client/components/Application/store.ts

```diff
@@ -194,45 +194,55 @@ export function makeStore(props: ApplicationProps) {
       }
     },
     adjustFile: async (name, type) => {
-      const { path, client, closeFile, loadFiles, selectFile } = get()
-      if (!path) return
-      await client.filePatch({ path, name, type })
-      set({ path: undefined })
+      const { paths, client, closeFile, loadFiles, selectFile } = get()
+      // only adjust file if one file is selected from menu
+      if (!paths || paths.length !== 1) return
+      await client.filePatch({ path: paths[0], name, type })
+      set({ paths: undefined })
       closeFile()
       await loadFiles()
-      await selectFile(path)
+      await selectFile(paths)
     },
-    copyFile: async (path, toPath) => {
+    copyFile: async (paths, toPath) => {
       const { client, onFileCreate } = get()
-      const result = await client.fileCopy({ path, toPath, deduplicate: true })
-      onFileCreate([result.path])
+      for (const path of paths) {
+        const result = await client.fileCopy({ path, toPath, deduplicate: true })
+        onFileCreate([result.path])
+      }
     },
-    deleteFile: async (path) => {
+    deleteFile: async (paths) => {
       const { client, onFileDelete } = get()
-      await client.fileDelete({ path })
-      onFileDelete(path)
+      for (const path of paths) {
+        await client.fileDelete({ path })
+        onFileDelete(path)
+      }
     },
     moveFile: async (path, toPath) => {
       const { client, onFileCreate } = get()
       const result = await client.fileMove({ path, toPath, deduplicate: true })
       onFileCreate([result.path])
     },
     locateFile: async (path) => {
-      set({ path })
+      set({ paths: [path] })
       set({ fileEvent: { type: 'locate', paths: [path] } })
       await delay(500)
       set({ fileEvent: undefined })
     },
-    selectFile: async (newPath) => {
-      const { path, record, openFile } = get()
-      if (path === newPath) return
-      set({ path: newPath })
-      if (!newPath) return
-      if (record?.path === newPath) return
+    selectFile: async (newPaths) => {
+      console.log('selectFile called with', newPaths)
+      const isSingleFile = newPaths && newPaths.length === 1
+      const { paths, record, openFile } = get()
+      if (paths === newPaths) return
+      set({ paths: newPaths })
+      if (!newPaths) return
+      if (record?.path === newPaths[0]) return
```

Comment:
not sure what this is doing. using only the first path just in case. Is it for the history? Do we need to modify the record to add all the selected paths?